cmd/tegola/cmd/root.go

```diff
@@ -152,3 +156,124 @@ func initConfig(configFile string, cacheRequired bool, logLevel string, logger s
 	atlas.SetObservability(observer)
 	return nil
 }
+
+// initProviders translate provider config from a TOML file into usable Provider objects.
+func initProviders(providersConfig []env.Dict, maps []provider.Map) (map[string]provider.TilerUnion, error) {
+	// first convert []env.Map -> []dict.Dicter
+	provArr := make([]dict.Dicter, len(providersConfig))
+	for i := range provArr {
+		provArr[i] = providersConfig[i]
+	}
+
+	providers, err := register.Providers(provArr, conf.Maps)
+	if err != nil {
+		return nil, fmt.Errorf("could not register providers: %v", err)
+	}
+
+	return providers, nil
+}
+
+// initMaps registers maps with Atlas to be ready for service.
+func initMaps(maps []provider.Map, providers map[string]provider.TilerUnion) error {
+	if err := register.Maps(nil, maps, providers); err != nil {
+		return fmt.Errorf("could not register maps: %v", err)
+	}
+
+	return nil
+}
+
+// initAppConfigSource sets up an additional configuration source for "apps" (groups of providers and maps) to be loaded and unloaded on-the-fly.
+func initAppConfigSource(conf config.Config) error {
+	// Get the config source type. If none, return.
+	val, err := conf.AppConfigSource.String("type", nil)
+	if err != nil || val == "" {
+		return nil
```

Comment:
Why are we not passing up the error here? Did you mean to swallow the error here?

Reply:
Yes, the swallowing is intentional. `Dict.String()` will return an error if the key is not present and no default is provided. This is an optional field with no default, so `Dict` is going to return errors when nothing erroneous has happened.

Reply:
If that's the case, ignore the error and check val. 
e.g.
```go

// don't care if an optional "type" entry does not exist in the config.
val, _ := conf.AppConfigSource.String("type",nil)
if val == "" {
   return nil
}
```